addons/account/models/account_move.py

```diff
@@ -2569,12 +2573,22 @@ def _autopost_draft_entries(self):
         It is used to post entries such as those created by the module
         account_asset.
         '''
+        try:
+            batch_size = int(self.env['ir.config_parameter'].sudo().get_param('account.autopost_batch_size')) or 1000
+        except ValueError as e:
+            batch_size = 1000
+            _logger.warning("Auto-post journal entry take default batch size because of the wrong value set in system parameter for key 'account.autopost_batch_size'")
+
         records = self.search([
             ('state', '=', 'draft'),
             ('date', '<=', fields.Date.today()),
             ('auto_post', '=', True),
         ])
-        records.post()
+        while records:
+            batch_records = records[batch_size]
+            batch_records.post()
+            records = records - batch_records
+            self._cr.commit()
```

Comment:
Like suggested twice by @fmdl 

Something like this is enough
```python
for recs in self.env.cr.split_for_in_conditions(records):
    recs.post()
    self.env.cr.commit()
```

Reply:
sorry for that, and adopted now.